pkg/plugins/generic/generic_plugin.go

```diff
@@ -103,6 +104,23 @@ func (p *GenericPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkN
 		if !found {
 			needDrain = true
 		}
+		if len(in.SriovMode) > 0 {
+			if in.SriovMode != "legacy" && in.SriovMode != "switchdev" {
+				return false, false, fmt.Errorf("generic-plugin OnNodeStateChange(): "+
+					"not supported SR-IOV mode %s for device %s, supported modes [legacy, switchdev]", in.PciAddress, in.SriovMode)
+			}
+			found = false
+			var ifaceOldStatus sriovnetworkv1.InterfaceExt
+			for _, ifaceOldStatus = range old.Status.Interfaces {
+				if ifaceOldStatus.PciAddress == in.PciAddress {
+					found = true
+					break
+				}
+			}
+			if !found || ifaceOldStatus.SriovMode != in.SriovMode {
+				needDrain = true
```

Comment:
Does this change require to drain the node?

Reply:
@adrianchiris moved it to new function